mc-sema/cfgToLLVM/JumpTables.cpp

```diff
@@ -342,6 +342,57 @@ static void doJumpTableViaSwitchReg(
 
 }
 
+void doJumpOffsetTableViaSwitchReg(
+        BasicBlock *& block, 
+        InstPtr ip, 
+        Value *regVal,
+        BasicBlock *&default_block,
+        Value *data_location,
+        MCSOffsetTablePtr ot_ptr)
+{
+
+    llvm::errs() << __FUNCTION__ << ": Doing jump offset table via switch(reg)\n";
+    Function *F = block->getParent();
+    
+
+    // create a default block that just traps
+    default_block = 
+        BasicBlock::Create(block->getContext(), "", block->getParent(), 0);
+    // end default block
+
+    const std::vector< std::pair<VA,VA> > &offset_dest = ot_ptr->getConstTable();
+    std::unordered_map<VA,VA> uniq_blocks;
```

Comment:
Suggest `std::map`. It's possible that the implicit ordering of keys that you'll get will help the LLVM optimizer to do better code gen with sorted switch cases. Alternatively, do `uniq_blocks.reserve(...)` to pre-allocate what's needed for the hash table.